### PR TITLE
WIP: add redis prior queue

### PR DESCRIPTION
**Pull request by git-hulk**
TODOs:

- [x] Check Redis v2 metrics were recorded as expected

- [ ]  The order of tasks with the same priority should be FIFO(it seems difficult while Redis zest would rank with the member)

### CODE REVIEW

File: engine/redis_v2/deadletter.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+)
+
+const (
+	LUA_D_RESPAWN = `
+local deadletter = KEYS[1]
+local queue = KEYS[2]
+local poolPrefix = KEYS[3]
+local limit = tonumber(ARGV[1])
+local respawnTTL = tonumber(ARGV[2])
+local score = tonumber(ARGV[2])
```
Comment (liguangbo) [inline engine/redis_v2/deadletter.go:19]: ARGV[2] => ARGV[3]

File: engine/redis_v2/deadletter.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+var (
+	_lua_respawn_deadletter_sha string
+	_lua_delete_deadletter_sha  string
+)
+
+// Because the DeadLetter is not like Timer which is a singleton,
+// DeadLetters are transient objects like Queue. So we have to preload
+// the lua scripts separately.
+func PreloadDeadLetterLuaScript(redis *redis_v1.RedisInstance) error {
```
Comment (liguangbo) [inline engine/redis_v2/deadletter.go:63]: *redis_v1.RedisInstance 这个要不要挪到engine里面去，只是通用包一层gp-redis。避免v2调用v1

File: server/handlers/queue.go [hunk 3]
```diff
@@ -179,7 +185,7 @@ func PublishBulk(c *gin.Context) {
 
 	jobIDs := make([]string, 0)
 	for _, job := range jobs {
-		jobID, err := e.Publish(namespace, queue, job, uint32(ttlSecond), uint32(delaySecond), uint16(tries))
+		jobID, err := e.Publish(namespace, queue, job, uint32(ttlSecond), uint32(delaySecond), uint16(tries), 0)
```
Comment (liguangbo) [inline server/handlers/queue.go:188]: 记录一下，批量暂未提供优先级

File: engine/redis_v2/deadletter.go (hunk 1 above)
Comment (git-hulk) [reply engine/redis_v2/deadletter.go:19]: good catch, 这个单元测试刚好也没有覆盖 time score，一起补充一下

**Review by liguangbo — DISMISSED** [review]
LGTM so far, brilliant!

cc @chensunny

File: engine/redis_v2/deadletter.go (hunk 2 above)
Comment (git-hulk) [reply engine/redis_v2/deadletter.go]: 这个就直接先 copy 一份到 v2, 之前考虑 v2 和 v1 有些代码可以复用，后面到最后其实不去动 v1 之后这个。

**Review by liguangbo — APPROVED** [review]
LGTM

**Pull request closed without merging**